resend/templates/_template.py

```diff
@@ -0,0 +1,95 @@
+"""Template and Variable type definitions."""
+
+from typing import Any, List, Literal, Union
+
+from typing_extensions import NotRequired, TypedDict
+
+
+class Variable(TypedDict):
+    """Template variable type.
+
+    Attributes:
+        key (str): The key of the variable. We recommend capitalizing the key (e.g. FIRST_NAME).
+        type (Literal["string", "number", "boolean", "object", "list"]): The type of the variable.
+        fallback_value (Any): The fallback value of the variable. Must match the type of the variable.
+            If no fallback value is provided, you must provide a value for the variable when sending
+            an email using the template. If object type is provided, you must include a fallback.
+    """
+
+    key: str
+    """The key of the variable. We recommend capitalizing the key (e.g. FIRST_NAME)."""
+
+    type: Literal["string", "number", "boolean", "object", "list"]
+    """The type of the variable."""
```

Comment:
We've decided to not add `boolean`, `object`, and `list` variable types yet (I noticed that the docs still didn't reflect this decision, sorry about that)
```suggestion
    Attributes:
        key (str): The key of the variable. We recommend capitalizing the key (e.g. FIRST_NAME).
        type (Literal["string", "number"]): The type of the variable.
        fallback_value (Any): The fallback value of the variable. Must match the type of the variable.
            If no fallback value is provided, you must provide a value for the variable when sending
            an email using the template.
    """

    key: str
    """The key of the variable. We recommend capitalizing the key (e.g. FIRST_NAME)."""

    type: Literal["string", "number"]
    """The type of the variable."""
```